src/agentlab/agents/hint_use_agent/generic_agent_prompt.py

```diff
@@ -0,0 +1,411 @@
+"""
+Prompt builder for GenericAgent
+
+It is based on the dynamic_prompting module from the agentlab package.
+"""
+
+import json
+import logging
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Literal
+
+import pandas as pd
+from agentlab.agents import dynamic_prompting as dp
+from agentlab.agents.tool_use_agent.tool_use_agent import HintsSource
+from agentlab.llm.chat_api import ChatModel
+from agentlab.llm.llm_utils import HumanMessage, parse_html_tags_raise
+from browsergym.core.action.base import AbstractActionSet
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class GenericPromptFlags(dp.Flags):
+    """
+    A class to represent various flags used to control features in an application.
+
+    Attributes:
+        use_plan (bool): Ask the LLM to provide a plan.
+        use_criticise (bool): Ask the LLM to first draft and criticise the action before producing it.
+        use_thinking (bool): Enable a chain of thoughts.
+        use_concrete_example (bool): Use a concrete example of the answer in the prompt for a generic task.
+        use_abstract_example (bool): Use an abstract example of the answer in the prompt.
+        use_hints (bool): Add some human-engineered hints to the prompt.
+        use_task_hint (bool): Enable task-specific hints from hint database.
+        hint_db_path (str): Path to the hint database file.
+        enable_chat (bool): Enable chat mode, where the agent can interact with the user.
+        max_prompt_tokens (int): Maximum number of tokens allowed in the prompt.
+        be_cautious (bool): Instruct the agent to be cautious about its actions.
+        extra_instructions (Optional[str]): Extra instructions to provide to the agent.
+        add_missparsed_messages (bool): When retrying, add the missparsed messages to the prompt.
+        flag_group (Optional[str]): Group of flags used.
+    """
+
+    obs: dp.ObsFlags
+    action: dp.ActionFlags
+    use_plan: bool = False  #
+    use_criticise: bool = False  #
+    use_thinking: bool = False
+    use_memory: bool = False  #
+    use_concrete_example: bool = True
+    use_abstract_example: bool = False
+    use_hints: bool = False
+    use_task_hint: bool = False
+    enable_chat: bool = False
+    max_prompt_tokens: int = None
+    be_cautious: bool = True
+    extra_instructions: str | None = None
+    add_missparsed_messages: bool = True
+    max_trunc_itr: int = 20
+    flag_group: str = None
+
+    # hint related
+    use_task_hint: bool = False
+    hint_type: str = "docs"
+    hint_index_type: str = "sparse"
+    hint_query_type: str = "direct"
+    hint_index_path: str = "indexes/servicenow-docs-bm25"
+    hint_retriever_path: str = "google/embeddinggemma-300m"
```

Comment:
I think this is missing some params such as hint_retrieval_model, hint_level, hint db path.